kornia/enhance/adjust.py

```diff
@@ -633,69 +633,116 @@ def _blend_one(input1: torch.Tensor, input2: torch.Tensor, factor: torch.Tensor)
     return torch.clamp(res, 0, 1)
 
 
-def _build_lut(histo, step):
-    # Compute the cumulative sum, shifting by step // 2
-    # and then normalization by step.
-    lut = (torch.cumsum(histo, 0) + (step // 2)) // step
-    # Shift lut, prepending with 0.
-    lut = torch.cat([torch.zeros(1, device=lut.device, dtype=lut.dtype), lut[:-1]])
-    # Clip the counts to be in range.  This is done
-    # in the C code for image.point.
-    return torch.clamp(lut, 0, 255)
-
-
-# Code taken from: https://github.com/pytorch/vision/pull/796
-def _scale_channel(im: torch.Tensor) -> torch.Tensor:
-    r"""Scale the data in the channel to implement equalize.
+def _equalize_channels(
+    im: torch.Tensor,
+    bins: Optional[int] = None,
+    minimum: Optional[Union[float, int]] = None,
+    maximum: Optional[Union[float, int]] = None,
+    dtype=None,
```

Comment:
I think device and dtype shall be inferred from the input tensor.

Reply:
the result type will be floating point even for integer inputs. these are common for volumetric medical images. it's more efficient to skip the intermediate conversion of `int16` to `float32` and process integers directly... in this case I was dealing with 500mb-1gb int16 images that need to be scaled before windowing.

Reply:
In kornia, I think we are assuming all inputs as floats in [0, 1].

\cc @edgarriba

Reply:
fair enough, I'll remove this